src/storage/src/source/mysql/snapshot.rs

```diff
@@ -213,9 +215,29 @@ pub(crate) fn render<G: Scope<Timestamp = GtidPartition>>(
                     .await?;
 
                 trace!(%id, "timely-{worker_id} acquiring table locks: {lock_clauses}");
-                lock_conn
+                match lock_conn
                     .query_drop(format!("LOCK TABLES {lock_clauses}"))
-                    .await?;
+                    .await
+                {
+                    // Handle the case where a table we are snapshotting has been dropped or renamed.
+                    Err(mysql_async::Error::Server(mysql_async::ServerError {
+                        code,
+                        message,
+                        ..
+                    })) if code == 1146 => {
```

Comment:
Let's lift this code up to a constant with a link to https://dev.mysql.com/doc/mysql-errors/8.0/en/server-error-reference.html#error_er_warning_not_complete_rollback